hazelcast/src/test/java/com/hazelcast/map/EntryProcessorOffloadableTest.java

```diff
@@ -928,6 +938,110 @@ public String getExecutorName() {
         }
     }
 
+    /**
+     * <pre>
+     * Given: Heart beats are configured to come each few seconds (i.e. one quarter of hazelcast.operation.call.timeout.millis
+     *        - set in the {@code getConfig()} method)
+     * When: An offloaded EntryProcessor takes a long time to run.
+     * Then: Heart beats are still coming during the task is offloaded.
+     * </pre>
+     *
+     * @see #getConfig()
+     * @see #HEARTBEATS_INTERVAL_SEC
+     */
+    @Test
+    public void testHeartBeatsCommingWhenEntryPropcessorOffloaded() throws InterruptedException, IOException {
+        final String key = generateKeyOwnedBy(instances[1]);
+        SimpleValue givenValue = new SimpleValue(1);
+
+        final IMap<Object, Object> map = instances[0].getMap(MAP_NAME);
+        map.put(key, givenValue);
+
+        final Address instance1Address = instances[1].getCluster().getLocalMember().getAddress();
+        final AtomicInteger heartBeatsCounter = new AtomicInteger(0);
+        final AtomicLong minDelta = new AtomicLong(Long.MAX_VALUE);
+        Thread hbMonitorThread = new Thread() {
+            public void run() {
+                OperationServiceImpl osImpl = (OperationServiceImpl) HazelcastTestSupport.getNodeEngineImpl(instances[0])
```

Comment:
If you anyway need two lines, I would split this up (easier to read than a long train wreck, especially with the cast):
```java
NodeEngine nodeEngine = HazelcastTestSupport.getNodeEngineImpl(instances[0]);
OperationServiceImpl osImpl = (OperationServiceImpl) nodeEngine.getOperationService();
```